### DISCUSSION TITLE
How do I force gltfpack to keep texture coords?

### DISCUSSION THREAD

**Discussion by bennlich** [zeux/meshoptimizer#348] [Q&A]
Hiya! I'm wondering if there's a flag I can pass to tell gltfpack not to throw away my UVs, even if my model has no textures.

Here's a simple cube exported from blender: https://drive.google.com/file/d/1SoBpm1-9eHL7PkB6ejYhQxT2RrbB-Pp0/view?usp=sharing

`gltf-transform inspect simple-cube.glb` shows that it has 3 attributes:

```
NORMAL:float32, POSITION:float32, TEXCOORD_0:float32
```
Then I pack it:

```
gltfpack -i simple-cube.glb -o packed-cube.glb
```

Now my NORMAL and POSITION attributes are quantized (WOO!), but the packed-cube no longer has a TEXCOORD attribute :(

```
NORMAL:int8, POSITION:uint16
```

Is there a flag I can pass so that the TEXCOORDs don't get tossed during packing?

**Comment by Rennzie** [comment]
I have a similar issue but using a custom attribute field to store featureIds on a per vertex basis. Wondering if there is a solution to keep the additional attributes around? @bennlich did you have any luck getting this working?

For context, I'm not doing any simplification with gltfpack. The warning that appears is:
```
Warning: ignoring unknown attribute _FEATURE_ID_0 in primitive i of mesh n
```

which comes from parseMeshesGltf https://github.com/zeux/meshoptimizer/blob/ad3d7e1d88e75f14669d3cb81e15f9abcf723dd4/gltf/parsegltf.cpp#L139

I'm interested in resolving this because we'd like to make use of the upcoming gltf extension [EXT_mesh_features](https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features#property-tables). I'm testing it without the extension for now i.e only using the attribute semantic. 

Would the behaviour change if there was an extension embedded in the gltf?

**Reply by zeux** [reply]
By itself it wouldn't; since gltfpack repacks the geometry, it will discard attributes that it doesn't know the format for. I was assuming that gltfpack can implement EXT_mesh_features as an *output* extension (that is, instead of preserving feature ids it could generate them), but maybe some simpler version of feature preservation would make sense as well depending on how complex the input data can be. This would need to be supported in [cgltf ](https://github.com/jkuhlmann/cgltf) first which gltfpack uses to parse glTF files.
